modules/core/include/opencv2/core/hal/intrin_cpp.hpp

```diff
@@ -1055,8 +1058,7 @@ template<typename _Tp, int n> inline v_reg<_Tp, n> v_select(const v_reg<_Tp, n>&
     for( int i = 0; i < n; i++ )
     {
         int_type m = Traits::reinterpret_int(mask.s[i]);
-        c.s[i] =  Traits::reinterpret_from_int((Traits::reinterpret_int(a.s[i]) & m)
-                                             | (Traits::reinterpret_int(b.s[i]) & ~m));
+        c.s[i] = m ? a.s[i] : b.s[i];
```

Comment:
BTW, This semantic is compatible with OpenCL: https://www.khronos.org/registry/OpenCL/sdk/1.2/docs/man/xhtml/select.html